test/setup.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-undef */
+import {jsdom} from 'jsdom';
+import global from 'global';
+
+// using jsDom to emulate a subset of a web browser to be useful for testing and
+// scraping real-world web applications
+// this is needed to enable using sinon in the node env
+// to test spies, stubs and mocks on function calls
+
+global.document = jsdom('');
+global.window = global.document.defaultView;
+Object.keys(global.document.defaultView).forEach((property) => {
```

Comment:
This is a lot of copying. When do we actually need this? What is deck.gl's layer manager's dependency on the DOM here? Does sinon not even work unless JSDom is available? That is a lot of dependencies just to be able to spy on a couple of function.

So, good if we could explain deck.gl's need for this more specifically in comments rather than just a generic statement about sinon.